### PR TITLE
Module 3 block 2 : Math Magicians: tests

### PR DESCRIPTION
**Pull request by aceDavon**
## We have Implemented the following tasks in this PR: :writing_hand: 

- [ ] Set up React Testing Library
- [ ] Follow the instructions in the official documentation
- [ ] Write unit tests for the files operate.js and calculate.js using Jest
- [ ] Create unit tests for all React components:
 - Use Jest snapshots to test the components
 - Use React Testing Library to simulate user interaction
- [ ] Run full tests suite using Jest

### CODE REVIEW

File: src/logic/logic.test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      operation: null,
+    };
+    const value = { total: 3, next: null, operation: '+' };
+    expect(calculate(data, '+')).toEqual(value);
+  });
+  test('Check operate for dividion by 0', () => {
+    const str = "Can't divide by 0.";
+    expect(operate(4, 0, '÷')).toBe(str);
+  });
+});
```
Comment (GzimAsani) [inline src/logic/logic.test.js]: - [ ] You should cover all the scenarios on operate and calculate components. Besides adding and division you should also test subtraction and multiplying

Comment (GzimAsani) [reply src/logic/logic.test.js]: - [ ] Please use `React Testing Library` to simulate user interaction.

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @aceDavon and @tarikbouari             :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your tests, you did most of the work +1: 
 
 ## Required Changes 

- [ ]  Update readme file and explain to the potential users how can they test your project
- [ ] Check the comments below :arrow_double_down: 




 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @aceDavon and @tarikbouari,

✔️ All tests are passing

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**